proto/tendermint/propagation/types.proto

```diff
@@ -8,6 +8,26 @@ import "tendermint/libs/bits/types.proto";
 import "tendermint/crypto/proof.proto";
 import "tendermint/types/types.proto";
 
+// TxMetaData contains all information needed to commit to a transaction. Start
+// and end are used to insert the transaction into the protobuf encoded version
+// of the block.
+message TxMetaData {
+  bytes  hash  = 1;
+  uint32 start = 2;
+  uint32 end   = 3;
+}
+
+// CompactBlock commits to the transaction included in a proposal. It enalbes
+// clients to reuse already downloaded blobs instead of gossiping them all again
+// during the block propagation.
+message CompactBlock {
+  bytes                     bp_hash     = 1;
+  repeated TxMetaData       blobs       = 2;
+  bytes                     signature   = 3;
+  tendermint.types.Proposal proposal    = 4;
+  uint32                    last_length = 5;
+}
```

Comment:
moved the proposal and the lastLen to the compact block instead

we also don't need the height or round since they'll be in the proposal. We could also move this back to the propagation reactor

Reply:
I guess it's cleaner having everything in the propagation reactor. good stuff 👍